src/vm/mod.rs

```diff
@@ -1,3 +1,115 @@
 // SPDX-License-Identifier: Apache-2.0
 
 mod linux;
+
+use crate::vm::linux::{
+    ioctl::{Cmd, CmdId},
+    types::{Capabilities, CpuidConfig, NR_CPUID_CONFIGS},
+};
+
+/// Handle to the KVM VM file descriptor
+pub struct Vm(pub kvm_ioctls::VmFd);
```

Comment:
```suggestion

use kvm_ioctls::VmFd;

/// Handle to the KVM VM file descriptor
pub struct Vm(pub VmFd);
```

I wonder if in this case, `Vm` could be a bit ambiguous, and would be better served named `TdxVm`?

Also, should `VmFd` be `pub`? We could make it private, and when the TDX launch process is finished, return it to a user's ownership. Curious to hear your thoughts.